barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -35,12 +35,12 @@ template <typename Curve> class IPA {
     static void compute_opening_proof(std::shared_ptr<CK> ck,
                                       const OpeningPair<Curve>& opening_pair,
                                       const Polynomial& polynomial,
-                                      BaseTranscript& transcript)
+                                      std::shared_ptr<BaseTranscript> transcript)
```

Comment:
It's a mouthful, but the more common idiom is to pass `const std::shared_ptr&` and avoid the ref copy. I don't think it matters for performance, but just pointing out the 'idiom aligns with performance' trait

Reply:
But don't we explicitly want the ref copy? Otherwise the smart pointer doesn't know how to be smart, no?

Reply:
![image](https://github.com/AztecProtocol/aztec-packages/assets/163993/6e81856d-cc64-4744-a493-1d9e2c5642f8)
https://stackoverflow.com/a/8741626
it's a pointer that can be refcounted. But normal RAII usage doesn't need the mechanism
enter a function +1 ref 
exit -1 ref
ref can never be below 0, therefore unneeded refcounts